packages/@aws-cdk/aws-amplify/README.md

```diff
@@ -231,6 +231,16 @@ const amplifyApp = new amplify.App(this, 'App', {
 });
 ```
 
+## Add a Framework property to your Branch
+
+Use the `framework` property on `BranchOptions` to pass in an optional value for the framework:
+
+```ts
+const amplifyApp = new amplify.App(this, 'MyApp', {
+  framework: 'ExampleFramework',
```

Comment:
Is any value here valid or are there specific frameworks that work?

Reply:
Thanks for your review Kendra! This is a question I've also been struggling to find the answer to -- the [CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-amplify-branch.html#cfn-amplify-branch-framework) has no information on what the framework property actually is and whether or not there are specific values that work (same with the [Amplify Docs](https://docs.aws.amazon.com/amplify/latest/APIReference/API_Branch.html)).

Reply:
@kaizencc Yes it did. Since I was able to successfully deploy with `test: TestFramework` it seems that this property can take in an arbitrary string, so there is not a list of specific frameworks that are valid; if this was the case, we would implement the property as an `enum` instead of a `string` @TheRealAmazonKendra

Reply:
Deploying with an arbitrary value doesn't always mean the service will work correctly once deployed. Give me a day or two to check with the service team on this.

Reply:
@kaizencc @TheRealAmazonKendra After speaking to David Calavera (SDM of Amplify) with Uday, we learned that the `framework` property is indeed meant to be an open text field where the user can enter any value (Amplify just ignores it if it does not recognize the framework).

However, we will also have to add a `platform` property in the `amplify.App` construct to solve the original problem from this issue - see the updated comments on the Github issue [here](https://github.com/aws/aws-cdk/issues/23325#issuecomment-1409411500) for full context.

Reply:
So, I've been doing a deep dive on this module because it's on our list to stabilize and it seems that the frameworks can't actually be any value. The template might deploy, but the branch won't deploy within the app. We should discover what the combinations of platforms and frameworks are valid and update the contract accordingly. This may require a much larger set of changes that you intended to take on here, though.

Feel free to hit me up and chat about this. If it does require sweeping and/or breaking changes, I might just accept this change for the moment so that we unblock customers on this, even if I will end up altering the contract.

Reply:
Still, the testing comments stand.

Reply:
Hi, PM here from the Amplify team.  I can confirm that the `framework` property in that it is an open text field for any value.  The detection around how to deploy the application happens based on the `platform` property.  `WEB` is for SPAs or HTML based applications and `WEB_COMPUTE` is for Next.js SSR only, at the moment.

Personally, I would approve this framework example test.